fn/concurrency.go

```diff
@@ -32,3 +34,43 @@ func ParSlice[V any](ctx context.Context, s []V, f ErrFunc[V]) error {
 
 	return errGroup.Wait()
 }
+
+// ParSliceErrCollect can be used to execute a function on each element of a
+// slice in parallel. This function is fully blocking and will wait for all
+// goroutines to finish. Error will be collected and returned as a map of slice
```

Comment:
"will wait for all goroutines to finish.", except for when the context is canceled from the outside, right?